client/core/core.go

```diff
@@ -497,7 +520,7 @@ func (c *Core) Exchanges() map[string]*Exchange {
 			FeePending:    dc.acct.feePending(),
 			Connected:     dc.connected,
 			ConfsRequired: uint32(dc.cfg.RegFeeConfirms),
-			RegConfirms:   dc.regConfirms,
+			RegConfirms:   dc.getRegConfirms(),
```

Comment:
When code accesses dereferences this `Exchange.RegConfirms` pointer, it's accessing `dexConnection.regConfirms` because`getRegConfirms` does `return &dc.regConfirms`.  So this is basically sidestepping the locking as when another goroutine calls `dc.setRegConfirms` it is not locked anymore.

You need to either have `getRegConfirms` return a plain `uint32` or make a copy inside the function and return a pointer to the copy (`confs := dc.regConfirms; return &confs`).

Reply:
Makes sense! I changed it to return the copy instead.